include/Quiccir/Interfaces/FoldTensorCastIntoConsumerOpInterface.td

```diff
@@ -0,0 +1,26 @@
+//===- FoldTensorCastIntoConsumerOpInterface.td - Fold Cast Interface -*- tablegen -==//
+//
+//   Copyright (c) 2024,
+//   Earth and Planetary Magnetism group, ETH Zurich
+//
+//===---------------------------------------------------------------------===//
+//
+// Defines the operations of the operations that can fold a tensor producer.
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef QUICCIR_INTERFACES_FOLDTENSORCASTINTOCONSUMEROPINTERFACE
```

Comment:
Do you remove the file extension from the header guard on purpose?

Reply:
I just followed the llvm convention, guards in `*.td` files have no extensions in the name